great_ape_safe/ape_api/badger.py

```diff
@@ -25,32 +25,39 @@ class Badger():
     def __init__(self, safe):
         self.safe = safe
 
+        contract_registry = registry()
+
         # tokens
         self.badger = interface.IBadger(
-            registry.eth.treasury_tokens.BADGER,
+            contract_registry.treasury_tokens.BADGER,
             owner=self.safe.account
         )
 
         # contracts
         self.tree = interface.IBadgerTreeV2(
-            registry.eth.badger_wallets.badgertree, owner=self.safe.account
-        )
-        self.strat_bvecvx = interface.IVestedCvx(
-            registry.eth.strategies['native.vestedCVX'],
-            owner=self.safe.account
-        )
-        self.timelock = safe.contract(
-            registry.eth.governance_timelock
+            contract_registry.badger_wallets.badgertree, owner=self.safe.account
         )
+
+        if chain.id == 1:
+            self.strat_bvecvx = interface.IVestedCvx(
+                contract_registry.strategies['native.vestedCVX'],
+                owner=self.safe.account
+            )
+            self.timelock = safe.contract(
+                contract_registry.governance_timelock
+            )
+            self.bribes_processor = interface.IBribesProcessor(
+                contract_registry.bribes_processor, owner=self.safe.account
+            )
+
         self.registry = interface.IBadgerRegistry(
-            registry.eth.registry, owner=self.safe.account
+            contract_registry.registry, owner=self.safe.account
         )
-        self.bribes_processor = interface.IBribesProcessor(
-            registry.eth.bribes_processor, owner=self.safe.account
+        self.registryV2 = interface.IBadgerRegistryV2(
+            contract_registry.registryV2, owner=self.safe.account
         )
```

Comment:
yea i agree with @petrovska-petro here, this now only works on chains for which `registryV2` is present in the address book (currently only fantom).

temp solution:
```suggestion
        if chain.id == 250:
            self.registryV2 = interface.IBadgerRegistryV2(
                contract_registry.registryV2, owner=self.safe.account
            )
```

Reply:
i've gone for a bit more of a completed approach, please have a look